sycl/test-e2e/Matrix/joint_matrix_prefetch_impl.hpp

```diff
@@ -0,0 +1,94 @@
+#define TM 8
+#define TK 16
+
+template <typename T1, typename T2, size_t M, size_t N, size_t K,
+          unsigned int vnniFactor>
+void matrix_multiply(T1 *C, T2 *A, T2 *vnniB, T2 *B, queue &q) {
+  size_t NDRangeM = M / TM;
+  size_t NDRangeN = N / TN;
+  auto pA = address_space_cast<sycl::access::address_space::global_space,
+                               sycl::access::decorated::no>(A);
+  auto pB = address_space_cast<sycl::access::address_space::global_space,
+                               sycl::access::decorated::no>(vnniB);
+  auto pC = address_space_cast<sycl::access::address_space::global_space,
+                               sycl::access::decorated::no>(C);
+  q.submit([&](handler &cgh) {
+     cgh.parallel_for<class imatrix>(
+         nd_range<2>({NDRangeM, NDRangeN * SG_SZ}, {1, 1 * SG_SZ}),
+         [=](nd_item<2> spmd_item) [[intel::reqd_sub_group_size(SG_SZ)]]
+
+         {
+           const auto global_idx = spmd_item.get_global_id(0);
+           const auto global_idy = spmd_item.get_global_id(1);
+           const auto sg_startx = global_idx - spmd_item.get_local_id(0);
+           const auto sg_starty = global_idy - spmd_item.get_local_id(1);
+
+           sub_group sg = spmd_item.get_sub_group();
+           joint_matrix<sub_group, bfloat16, use::a, TM, TK, layout::row_major>
+               sub_a;
+           // For B, we assume B has been already VNNIed.
+           joint_matrix<sub_group, bfloat16, use::b, TK, TN,
+                        layout::ext_intel_packed>
+               sub_b;
+           joint_matrix<sub_group, float, use::accumulator, TM, TN> sub_c;
+
+           joint_matrix_prefetch(sg, A, (sg_startx * TM) * K, layout::row_major,
+                                 TM, TK,
+                                 syclex::properties{syclex::prefetch_hint_L1});
+           joint_matrix_prefetch(
+               sg, vnniB + sg_starty / SG_SZ * TN * vnniFactor, N * vnniFactor,
+               layout::ext_intel_packed, TK / vnniFactor, TN * vnniFactor,
+               syclex::properties{syclex::prefetch_hint_L1});
+           joint_matrix_prefetch(sg, B + sg_starty / SG_SZ * TN, N,
```

Comment:
Why do we need to prefetch B, if we are using vnniB only in this kernel?

Reply:
I will change the code so both are used